foundry/applications/cameraViews.d.ts

```diff
@@ -1 +1,180 @@
+   * @param userId - The ID of the User entity
+   */
+  getUserVideoElement(userId: string): HTMLVideoElement | null;
+
+  /**
+   * Sets whether a user is currently speaking or not
+   *
+   * @param userId   - The ID of the user
+   * @param speaking - Whether the user is speaking
+   */
+  setUserIsSpeaking(userId: string, speaking: boolean): void;
+
+  /**
+   * Extend the render logic to first check whether a render is necessary based on the context
+   * If a specific context was provided, make sure an update to the navigation is necessary before rendering
+   */
+  render(force?: boolean, context?: Application.RenderOptions): Application['render'];
+
+  /** @override */
+  protected _render(force?: boolean, options?: Application.RenderOptions): Promise<void>;
+
+  /**
+   * @param options - (unused)
+   * @override
+   */
+  getData(options?: Application.RenderOptions): CameraViews.Data;
+
+  maxZ: number;
+
+  /**
+   * Prepare rendering data for a single user
+   */
+  protected _getDataForUser(userId: string, settings: AVSettings.UserSettings): CameraViews.Data.User | null;
+
+  /**
+   * A custom sorting function that orders/arranges the user display frames
+   */
+  protected static _sortUsers(a: CameraViews.Data.User, b: CameraViews.Data.User): number;
+
+  /** @override */
+  activateListeners(html: JQuery): void;
+
+  /**
+   * On hover in a camera container, show/hide the controls.
+   * @param event - The original mouseover or mouseout hover event
+   */
+  protected _onCameraViewHover(event: JQuery.MouseEnterEvent | JQuery.MouseLeaveEvent): void;
+
+  /**
+   * On clicking on a toggle, disable/enable the audio or video stream.
+   * @param event - The originating click event
+   */
+  protected _onClickControl(event: JQuery.ClickEvent): Promise<void>;
+
+  /**
+   * Change volume control for a stream
+   * @param event - The originating change event from interaction with the range input
+   */
+  protected _onVolumeChange(event: JQuery.ChangeEvent): void;
+
+  /**
+   * Dynamically refresh the state of a single camera view
+   * @param view - The view container div
+   */
+  protected _refreshView(view: HTMLElement): void;
+
+  /**
+   * Render changes needed to the PlayerList ui.
+   * Show/Hide players depending on option.
+   */
+  protected _setPlayerListVisibility(): void;
+
+  /**
+   * Get the icon class that should be used for various action buttons with different toggled states.
+   * The returned icon should represent the visual status of the NEXT state (not the CURRENT state).
+   *
+   * @param action - The named av-control button action
+   * @param state  - The CURRENT action state.
+   * @returns The icon that represents the NEXT action state.
+   */
+  protected _getToggleIcon<A extends keyof CameraViews.ToggleIcons, S extends boolean>(
+    action: A,
+    state: S
+  ): CameraViews.ToggleIcons[A][S extends true ? 1 : 0];
+  protected _getToggleIcon(action: string, state?: any): null;
```

Comment:
shouldn't `state` be a boolean?

Reply:
No that's on purpose in the second overload, because the code never gets to checking that parameter, if the action isn't one of the predetermined ones.